packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -313,40 +329,47 @@ export const transformPlotlyJsonToVSBCProps = (
-      if (!mapXToDataPoints[x]) {
-        mapXToDataPoints[x] = { xAxisPoint: isXYearCategory ? x.toString() : x, chartData: [], lineData: [] };
-      }
-      const legend: string = legends[index1];
-      // resolve color for each legend's bars from the extracted colors
-      const color = resolveColor(extractedBarColors, index1, legend, colorMap, isDarkTheme);
-      const yVal: number = (series.y?.[index2] as number) ?? 0;
-      if (series.type === 'bar') {
-        mapXToDataPoints[x].chartData.push({
-          legend,
-          data: yVal,
-          color,
-        });
-        yMaxValue = Math.max(yMaxValue, yVal);
-      } else if (series.type === 'scatter' || !!fallbackVSBC) {
-        const lineColor = resolveColor(extractedLineColors, index1, legend, colorMap, isDarkTheme);
-        const lineOptions = getLineOptions(series.line);
-        const dashType = series.line?.dash || 'solid';
-        const legendShape =
-          dashType === 'dot' || dashType === 'dash' || dashType === 'dashdot' ? 'dottedLine' : 'default';
-        mapXToDataPoints[x].lineData!.push({
-          legend,
-          legendShape,
-          y: yVal,
-          color: lineColor,
-          ...(lineOptions ? { lineOptions } : {}),
-          useSecondaryYScale: usesSecondaryYScale(series),
-        });
-        if (!usesSecondaryYScale(series)) {
+
+    const validXYRanges = getValidXYRanges(series);
+    validXYRanges.forEach(([rangeStart, rangeEnd], rangeIdx) => {
+      const rangeXValues = series.x!.slice(rangeStart, rangeEnd);
+      const rangeYValues = series.y!.slice(rangeStart, rangeEnd);
+
+      (rangeXValues as Datum[]).forEach((x: string | number, index2: number) => {
+        if (!mapXToDataPoints[x]) {
+          mapXToDataPoints[x] = { xAxisPoint: isXYearCategory ? x.toString() : x, chartData: [], lineData: [] };
+        }
+        const legend: string = legends[index1];
+        // resolve color for each legend's bars from the extracted colors
+        const color = resolveColor(extractedBarColors, index1, legend, colorMap, isDarkTheme);
+        const yVal: number = rangeYValues[index2] as number;
+        if (series.type === 'bar') {
+          mapXToDataPoints[x].chartData.push({
+            legend,
+            data: yVal,
+            color,
+          });
           yMaxValue = Math.max(yMaxValue, yVal);
-          yMinValue = Math.min(yMinValue, yVal);
+        } else if (series.type === 'scatter' || !!fallbackVSBC) {
+          const lineColor = resolveColor(extractedLineColors, index1, legend, colorMap, isDarkTheme);
+          const lineOptions = getLineOptions(series.line);
+          const dashType = series.line?.dash || 'solid';
+          const legendShape =
+            dashType === 'dot' || dashType === 'dash' || dashType === 'dashdot' ? 'dottedLine' : 'default';
+          mapXToDataPoints[x].lineData!.push({
+            legend: legend + (validXYRanges.length > 1 ? `.${rangeIdx + 1}` : ''),
```

Comment:
>${rangeIdx + 1} [](http://example.com/codeflow?start=59&length=15)

are we required to create new legend if the series is the same

Reply:
Yes. VSBC accepts data points grouped by x-axis values. Internally, it groups line points by legend before rendering, which causes the line gaps to disappear when the legends are the same.